pkgs/development/compilers/terra/default.nix

```diff
@@ -1,61 +1,70 @@
 { lib, stdenv, fetchurl, fetchFromGitHub
-, llvmPackages, ncurses, lua
+, llvmPackages, ncurses, cmake, libxml2, symlinkJoin
 }:
 
 let
-  luajitArchive = "LuaJIT-2.0.5.tar.gz";
+  luajitRev = "9143e86498436892cb4316550be4d45b68a61224";
+  luajitArchive = "LuaJIT-${luajitRev}.tar.gz";
   luajitSrc = fetchurl {
-    url    = "http://luajit.org/download/${luajitArchive}";
-    sha256 = "0yg9q4q6v028bgh85317ykc9whgxgysp76qzaqgq55y6jy11yjw7";
+    url = "https://github.com/LuaJIT/LuaJIT/archive/${luajitRev}.tar.gz";
+    sha256 = "0kasmyk40ic4b9dwd4wixm0qk10l88ardrfimwmq36yc5dhnizmy";
   };
 in
 stdenv.mkDerivation rec {
   pname = "terra";
-  version = "1.0.0pre1175_${builtins.substring 0 7 src.rev}";
+  version = "1.0.0-beta3";
 
   src = fetchFromGitHub {
-    owner  = "zdevito";
+    owner  = "terralang";
     repo   = "terra";
-    rev    = "ef6a75ffee15a30f3c74f4e6943851cfbc0fec3d";
-    sha256 = "0aky17vbv3d9zng34hp17p9zb00dbzwhvzsdjzrrqvk9lmyvix0s";
+    rev    = "release-${version}";
+    sha256 = "15ik32xnwyf3g57jvvaz24f6a18lv3a86341rzjbs30kd5045qzd";
   };
 
-  nativeBuildInputs = [ lua ];
-  buildInputs = with llvmPackages; [ llvm clang-unwrapped ncurses ];
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    (symlinkJoin {
+      name = "llvmClangMerged";
+      paths = with llvmPackages; [ llvm clang-unwrapped ];
+    })
+    ncurses
+    libxml2
+  ];
+
+  cmakeFlags = [
+    "-DHAS_TERRA_VERSION=0"
+    "-DTERRA_VERSION=${src.rev}"
+    "-DTERRA_LUA=luajit"
+  ];
 
   doCheck = true;
   enableParallelBuilding = true;
   hardeningDisable = [ "fortify" ];
   outputs = [ "bin" "dev" "out" "static" ];
 
-  patches = [ ./nix-cflags.patch ];
+  patches = [
+    # Should be removed as https://github.com/terralang/terra/pull/495 get merged and released.
+    ./get-compiler-from-envvar-fix-cpu-detection.patch
```

Comment:
Okay, Fundament has separated our `tcompiler.cpp` changes into a separate pull, https://github.com/terralang/terra/pull/496, which has just been merged. You should be able to remove this patch now, and point at that commit instead.

Reply:
https://github.com/terralang/terra/commit/713f5bacb95cb443b6a86947aa01e11febd8b49f and https://github.com/terralang/terra/commit/f37b92c55326679f0d9c0bd0041ac6680610a78d don't let me use this commit cleanly. I've just updated the comment.

Reply:
How so?

Reply:
They modify `tcompiler.cpp` and I can't apply  terralang/terra#496 whithout apply them first.

Reply:
I don't get it. What's stopping you from just pointing your fetch at https://github.com/terralang/terra/commit/8f3b656ff5345717fffcbd476c431167e1a6fe7c since https://github.com/terralang/terra/pull/496 has already been merged?

Reply:
Oh, I've got it. I was too fixated with starting off release-1.0.0-beta3 and then adding patches on top of it.